panic after adding source-acr-auth-id to acr task

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

After [applying a user assigned identity via "az acr task update --source-acr-auth-id ${identity}"](https://learn.microsoft.com/en-us/azure/container-registry/container-registry-rbac-abac-repository-permissions?tabs=azure-portal#attaching-an-identity-to-a-new-acr-task-for-source-registry-access) to an existing task, trying to run terraform plan for this tasks causes a panic in the provider.

### Terraform Version

OpenTofu 1.10.5

### AzureRM Provider Version

4.50.0

### Affected Resource(s)/Data Source(s)

azurerm_container_registry_task

### Terraform Configuration Files

```hcl
-
```

### Debug Output/Panic Output

```shell
Stack trace from the terraform-provider-azurerm plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x56ee5bb]

goroutine 181 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/containers.flattenSourceRegistryCredential(...)
	/home/runner/work/terraform-provider-azurerm/terraform-provider-azurerm/provider/internal/services/containers/container_registry_task_resource.go:1517
github.com/hashicorp/terraform-provider-azurerm/internal/services/containers.flattenRegistryTaskCredentials(_, {{0xc0055f0690, 0x11}, {0xc001d2a1b0, 0x8a}, {0xe324500, 0x0, 0x0}, {0xc0055ea598, 0x5}, ...})
	/home/runner/work/terraform-provider-azurerm/terraform-provider-azurerm/provider/internal/services/containers/container_registry_task_resource.go:1498 +0x7b
github.com/hashicorp/terraform-provider-azurerm/internal/services/containers.(*ContainerRegistryTaskResource).Read.ContainerRegistryTaskResource.Read.func1({0x8dcc200, 0xc001158cb0}, {0xc000452008, {0x8dcfc68, 0xc000f86cc0}, 0xc0056287e0, 0x0, {0x8dcfea0, 0xe324500}})
	/home/runner/work/terraform-provider-azurerm/terraform-provider-azurerm/provider/internal/services/containers/container_registry_task_resource.go:822 +0x978
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.(*ResourceWrapper).Resource.func3({0x8dcc200, 0xc001158cb0}, 0x8db1f70?, {0x7429560?, 0xc000452008?})
	/home/runner/work/terraform-provider-azurerm/terraform-provider-azurerm/provider/internal/sdk/wrapper_resource.go:69 +0x145
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.(*ResourceWrapper).Resource.(*ResourceWrapper).diagnosticsWrapper.diagnosticsWrapper.func14({0x8dcc200?, 0xc001158cb0?}, 0x45d964b800?, {0x7429560?, 0xc000452008?})
	/home/runner/work/terraform-provider-azurerm/terraform-provider-azurerm/provider/internal/sdk/wrapper_resource.go:218 +0x59
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc000d14700, {0x8dcc158, 0xc00563ac90}, 0xc0056287e0, {0x7429560, 0xc000452008})
	/home/runner/work/terraform-provider-azurerm/terraform-provider-azurerm/provider/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:866 +0x119
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000d14700, {0x8dcc158, 0xc00563ac90}, 0xc0017ca410, {0x7429560, 0xc000452008})
	/home/runner/work/terraform-provider-azurerm/terraform-provider-azurerm/provider/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:1162 +0x52a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc000c21770, {0x8dcc158?, 0xc00563abd0?}, 0xc0057247d0)
	/home/runner/work/terraform-provider-azurerm/terraform-provider-azurerm/provider/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:915 +0xb6d
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ReadResource(0xc0008a59e0, {0x8dcc158?, 0xc0057098f0?}, 0xc0057247d0)
	/home/runner/work/terraform-provider-azurerm/terraform-provider-azurerm/provider/vendor/github.com/hashicorp/terraform-plugin-mux/tf5muxserver/mux_server_ReadResource.go:35 +0x190
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc0017b2000, {0x8dcc158?, 0xc005708de0?}, 0xc0014a03f0)
	/home/runner/work/terraform-provider-azurerm/terraform-provider-azurerm/provider/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server/server.go:862 +0x2f3
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x83bdf40, 0xc0017b2000}, {0x8dcc158, 0xc005708de0}, 0xc005710200, 0x0)
	/home/runner/work/terraform-provider-azurerm/terraform-provider-azurerm/provider/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:753 +0x1a9
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000cb2800, {0x8dcc158, 0xc005708d50}, 0xc00157fc20, 0xc001cddbc0, 0xe2f0918, 0x0)
	/home/runner/work/terraform-provider-azurerm/terraform-provider-azurerm/provider/vendor/google.golang.org/grpc/server.go:1431 +0x1036
google.golang.org/grpc.(*Server).handleStream(0xc000cb2800, {0x8dd0218, 0xc001bfa000}, 0xc00157fc20)
	/home/runner/work/terraform-provider-azurerm/terraform-provider-azurerm/provider/vendor/google.golang.org/grpc/server.go:1842 +0xb88
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	/home/runner/work/terraform-provider-azurerm/terraform-provider-azurerm/provider/vendor/google.golang.org/grpc/server.go:1061 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 55
	/home/runner/work/terraform-provider-azurerm/terraform-provider-azurerm/provider/vendor/google.golang.org/grpc/server.go:1072 +0x11d
```

### Expected Behaviour

_No response_

### Actual Behaviour

_No response_

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/containers/container_registry_task_resource.go
+++ b/internal/services/containers/container_registry_task_resource.go
@@ -1510,7 +1510,7 @@ func expandSourceRegistryCredential(input []SourceRegistryCredential) *tasks.Sou
 }
 
 func flattenSourceRegistryCredential(input *tasks.SourceRegistryCredentials) []SourceRegistryCredential {
-	if input == nil {
+	if input == nil || input.LoginMode == nil {
 		return nil
 	}
 

